File: 541/reverse-string-ii.py
class Solution:
    def reverseStr(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        tmp = []
        ans = []
        for i in range(len(s)):
            if (i % (2*k) < k):
                tmp.append(s[i])
                continue
            while len(tmp) > 0:
                ans.append(tmp.pop())
            ans.append(s[i])

        while len(tmp) > 0:
            ans.append(tmp.pop())

        return ''.join(ans)

File: 541/reverse-string-ii.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var reverseStr = function (s, k) {
    // if (k == 1)
    //     return s;
    let tmp = [];
    let ans = [];
    for (let i = 0, len = s.length; i < len; i++) {
        if (i % (2 * k) < k) {
            tmp.push(s[i]);
            continue;
        }
        while (tmp.length) {
            ans.push(tmp.pop());
        }
        ans.push(s[i]);
    }
    while (tmp.length) {
        ans.push(tmp.pop());
    }
    return ans.join('');
};